File: SEPADirectDebit/src/main/java/com/braintreepayments/api/SEPADirectDebitTokenizeCallback.java
package com.braintreepayments.api;

import androidx.annotation.Nullable;

/**
 * Callback for receiving result of
 * {@link SEPADirectDebitClient#tokenize(SEPADirectDebitPaymentAuthResult, SEPADirectDebitTokenizeCallback)}.
 */
public interface SEPADirectDebitTokenizeCallback {

    /**
     * @param sepaDirectDebitNonce {@link SEPADirectDebitNonce}
     * @param error                an exception that occurred while processing a PayPal result
     */
    void onResult(@Nullable SEPADirectDebitNonce sepaDirectDebitNonce, @Nullable Exception error);
}

File: SEPADirectDebit/src/main/java/com/braintreepayments/api/SEPADirectDebitTokenizeCallback.kt
package com.braintreepayments.api

/**
 * Callback for receiving result of [SEPADirectDebitClient.tokenize]
 */
fun interface SEPADirectDebitTokenizeCallback {
    /**
     * @param sepaDirectDebitResult a success, failure, or cancel result from the SEPA Direct Debit
     * flow
     */
    fun onSEPADirectDebitResult(sepaDirectDebitResult: SEPADirectDebitResult)
}
